lsp-modeline.el

```diff
@@ -0,0 +1,268 @@
+    (propertize string
+                'help-echo (concat (format "Apply code actions %s\nmouse-1: " keybinding)
+                                   (if single-action?
+                                       first-action-string
+                                     "select from multiple code actions"))
+                'mouse-face 'mode-line-highlight
+                'local-map (make-mode-line-mouse-map
+                            'mouse-1 (lambda ()
+                                       (interactive)
+                                       (if single-action?
+                                           (lsp-execute-code-action (lsp-seq-first actions))
+                                         (lsp-execute-code-action (lsp--select-action actions))))))))
+
+(defun lsp--modeline-update-code-actions (actions)
+  "Update modeline with new code ACTIONS."
+  (when lsp-modeline-code-actions-kind-regex
+    (setq actions (seq-filter (-lambda ((&CodeAction :kind?))
+                                (or (not kind?)
+                                    (s-match lsp-modeline-code-actions-kind-regex kind?)))
+                              actions)))
+  (setq lsp-modeline--code-actions-string
+        (if (seq-empty-p actions) ""
+          (lsp-modeline--build-code-actions-string actions)))
+  (force-mode-line-update))
+
+(defun lsp-modeline--check-code-actions (&rest _)
+  "Request code actions to update modeline for given BUFFER."
+  (when (lsp-feature? "textDocument/codeAction")
+    (lsp-request-async
+     "textDocument/codeAction"
+     (lsp--text-document-code-action-params)
+     #'lsp--modeline-update-code-actions
+     :mode 'unchanged
+     :cancel-token :lsp-modeline-code-actions)))
+
+(defun lsp-modeline--enable-code-actions ()
+  "Enable code actions on modeline mode."
+  (when (and lsp-modeline-code-actions-enable
+             (lsp-feature? "textDocument/codeAction"))
+    (lsp-modeline-code-actions-mode 1)))
+
+(defun lsp-modeline--disable-code-actions ()
+  "Disable code actions on modeline mode."
+  (lsp-modeline-code-actions-mode -1))
+
+;;;###autoload
+(define-minor-mode lsp-modeline-code-actions-mode
+  "Toggle code actions on modeline."
+  :group 'lsp-mode
+  :global nil
+  :lighter ""
+  (cond
+   (lsp-modeline-code-actions-mode
+    (add-to-list 'global-mode-string '(t (:eval lsp-modeline--code-actions-string)))
+
+    (add-hook 'lsp-on-idle-hook 'lsp-modeline--check-code-actions nil t)
+    (add-hook 'lsp-configure-hook #'lsp-modeline--enable-code-actions nil t)
+    (add-hook 'lsp-unconfigure-hook #'lsp-modeline--disable-code-actions nil t))
+   (t
+    (remove-hook 'lsp-on-idle-hook 'lsp-modeline--check-code-actions t)
+    (remove-hook 'lsp-configure-hook #'lsp-modeline--enable-code-actions t)
+    (remove-hook 'lsp-unconfigure-hook #'lsp-modeline--disable-code-actions t)
+    (setq global-mode-string (remove '(t (:eval lsp-modeline--code-actions-string)) global-mode-string)))))
+
```

Comment:
too much space.

Reply:
It's the page break: 
![image](https://user-images.githubusercontent.com/7820865/88178969-8ebce880-cc01-11ea-9de1-1c6efcef4949.png)

Reply:
Ah, ok.